net/http/isacceptablepath.c

```diff
@@ -66,7 +67,8 @@ bool IsAcceptablePath(const char *data, size_t size) {
       x = '/';
     }
     if (y == '/') {
-      if (x == '.') return false;
+      if (x == '.' &&  // allow /.well-known/ in the first position
+         (p - data > 2 || !startswith(data, "/.well-known/"))) return false;
```

Comment:
You can't assume the string is nul-terminated here. Could you please check the full length?

Reply:
Correct; will do.